src/serializers/mod.rs

```diff
@@ -174,6 +183,14 @@ impl SchemaSerializer {
         Ok(py_bytes.into())
     }
 
+    pub fn __reduce__(&self, py: Python) -> PyResult<(PyObject, (PyObject, PyObject))> {
+        // Enables support for `cloudpickle` serialization.
+        Ok((
+            SchemaSerializer::type_object(py).to_object(py),
+            (self.py_schema.to_object(py), self.py_config.to_object(py)),
+        ))
+    }
```

Comment:
Here's my take:

```suggestion
    pub fn __reduce__(slf: &PyCell<Self>) -> PyResult<(&PyType, (PyObject, PyObject))> {
        let py = slf.py();
        let args = (slf.get().schema.to_object(py), slf.get().py_config.to_object(py));
        let cls = slf.get_type();
        Ok((cls, args))
    }
```

You'll need to update the `#[pyclass]` definition above for `SchemaSerializer` to include `#[pyclass(frozen)]`.

Justifications:

1. `slf: &PyCell<Self>` is equivalent to `self` Python object
2. `slf.get()` gives read access to the Rust data as long as it's `frozen`
3. I think we want to look up `type(self)` rather than hard-code `SchemaSerializer` type in case we allow subclassing later. `slf.get_type()` is a cleaner way to do it than trying to use `getattr`.

Reply:
Makes sense, thanks for the pointers, I'll update!